### DISCUSSION TITLE
[PorousFlow] why a small amount of rainfall flooded the model (PorousFlowSink)

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#22867] [Q&A Modules: General] [answered]
Hi guys, I'm working on the underground water modelling and want to add the rainfall into the model

so here is a simple model (1km x 1km x 0.5 km 20^3 mesh) with 15% porosity and perm of 1e-15 m2

here are the BCs: basically (rho g h) at the 4 sides + rain at the top

and the amount of rain basically is from a quick google search
"Australia has a very low annual average rainfall of 419 mm"

so 419 mm year-1 = 419* 1e-3 m3 m-2 year-1 = 419 kg m-2 year -1

note that my time unit is in YEARS

```
  [./ini_pp]
    type = ParsedFunction
    value  = 9.81*1000*(500-z)
  [../]

     [./water_grad_left]
       type = FunctionDirichletBC
       variable = porepressure
       boundary = 'left'
       function = ini_pp
       # save_in = fluxes_out
     [../]

     [./water_grad_right]
       type = FunctionDirichletBC
       variable = porepressure
       boundary = 'right'
       function = ini_pp
        # save_in = fluxes_out
     [../]

     [./water_grad_top]
       type = FunctionDirichletBC
       variable = porepressure
       boundary = 'top'
       function = ini_pp
        # save_in = fluxes_out
     [../]

     [./water_grad_bottom]
       type = FunctionDirichletBC
       variable = porepressure
       boundary = 'bottom'
       function = ini_pp
        # save_in = fluxes_out
     [../]

          [rainfall_recharge]
             type = PorousFlowSink
             fluid_phase = 0
             boundary = 'front'
             variable = porepressure
             flux_function = - 419
             save_in = fluxes_out
           []
```

simulation is run in steady-state, and here is the result:
![image](https://user-images.githubusercontent.com/79562439/205552802-1ce6c347-38ca-4bfa-adf9-e58635aae4aa.png)

but then when I change the perm to unrealistically high (1e-12 m2), this is what I got + the streamline move in a totally different way
![image](https://user-images.githubusercontent.com/79562439/205553093-4836e66c-2d70-47ab-b0a7-f5521a63b398.png)

feel like with an extremely high perm case more water is able to move out from the system (from top to bottom of the sides), but in a normal groundwater perm case, not all the water can move out from the system (streamline at the sides is minimal).

not sure if my PorousFlowSink flux_function is correct

@WilkAndy I'm tagging you here since you are the porousflow expert

Kind regards,
Traiwit

**Comment by WilkAndy** [comment, marked as answer]
Hi @Traiwit - there is nothing wrong with your `BCs` **if** you believe that 419mm/year is entering the groundwater system.  However, that is likely unphysical.  Recently, we quantified recharge to the Australia's groundwater system - have a look at https://doi.org/10.25919/waky-s739 .   There you will see values much less than 419mm/year.  I don't know what area you're looking at, but a value of 10mm/year is more likely.

The reason this is a lot smaller than the rainfall is that most water ends up as runoff, or is evapotranspirated.  If you are doing any reasonably realistic groundwater modelling, your model should likely include regional groundwater flow (can be set by using appropriate Dirichlet BCs, for instance) as well as evapotranspiration (i would use https://mooseframework.inl.gov/source/bcs/PorousFlowHalfCubicSink.html , and you can find estimates of the "pan evaporation" online - eg at the BOM http://www.bom.gov.au/climate/change/about/evap_averagemaps.shtml ).

**Reply by Traiwit** [reply]
Thanks for the answer @WilkAndy, if I understand correctly, IF i want to use only `PorousFlowSink` I need to consider: rainfall + evaporation and flow at the surface altogether right? and that would be 'surface recharge' data instead of just 'rainfall'

**Reply by WilkAndy** [reply]
Yes, that is true, @Traiwit , but your model will likely quite rapidly fill up, if only your "side" boundary conditions are letting water out.  Also, it will likely not be possible if your topography is non-horizontal, as you will have natural seep points at the bottom of valleys, where you should let water out by using a PorousFlowPiecewiseLinearSink.

**Reply by Traiwit** [reply]
Hi @WilkAndy 

Not sure how do I set `PorousFlowPiecewiseLinearSink` to the side to let water out WHILE maintaining the rho x g x h

for our real models, we define pwp at all the nodes on 4 sides including the mountain range from CSV > PiecewiseConstantFromCSV > BC.

![image](https://user-images.githubusercontent.com/79562439/205859315-d9411617-628c-4d99-ac85-05c2898cbb33.png)
 

-------------------------------------------------------------
Also nice that you brought up `the non-horizontal topol`, lets me explain the actual real model we are working on, it might not be related to what you mentioned, but you may find it interesting:

Yes, most of our topols are not completely flat, for example:
![image](https://user-images.githubusercontent.com/79562439/205852370-9c3616e7-c2b7-4b38-9e34-369bb5b71b9c.png)

where is orange surface is the pit with drainage BC
and the topol BC just set so the water table is 10m under the top of the system

```
[./drain]
type = PorousFlowPiecewiseLinearSinkBEH #exactly the same as PorousFlowPiecewiseLinearSink just add 1/element length to the equation
variable = porepressure
boundary = 'DRAINSRFPIT'
pt_vals = '0 1e9'
multipliers = '0 1e9'
flux_function = 1
v = min_dist
PT_shift = 0
fluid_phase = 0
use_mobility = true
[../]

[./topol_linear]
type = PorousFlowPiecewiseLinearSink
variable = porepressure
boundary = 'TOPOLSRF'
pt_vals = '-1e9 1e9'
multipliers = '-1e9 1e9'
flux_function = 1
v = min_dist
PT_shift = -98100 #10m below surface
fluid_phase = 0
[../]
```

It has been working fine just like that, BUT for this particular problem/mine it has a cave area where the perm is high compared to the rest of the model
![image](https://user-images.githubusercontent.com/79562439/205853982-c7984b96-d639-45de-becb-73acc38dc0d2.png)

here is the result if I use the `PorousFlowPiecewiseLinearSink `for topol 
![image](https://user-images.githubusercontent.com/79562439/205855528-dffd4f10-9d6c-424e-afb8-fac41cd5f608.png)

Hence we are trying to use `PorousFlowSink` with real data to deal with this problem instead. I tried `flux_function = -10` just to test it out, here is the result: (the system is drained too much but at the edge it still flooded with water)
![image](https://user-images.githubusercontent.com/79562439/205856130-c8469049-39b8-48af-85c6-4ceb28192974.png)
![image](https://user-images.githubusercontent.com/79562439/205856282-1c209040-fd70-4dab-9d0f-3a8fba40144f.png)

At this current stage, I still don't know what to do, might try mixing PorousFlowPiecewiseLinearSink and PorousFlowSink to balance them out.

Please let me know if you have any idea on how to deal with this problem, otherwise, it's okay I find it quite overwhelming too.

Thank you!
Traiwit

**Reply by WilkAndy** [reply]
I only meant applying the `PorousFlowPiecewiseLinearSink` on the top of the model, not on its sides.  So, just like you've done with your `drain` (but over the entire topography, not just the pit).  

Let's discuss your `topol_linear`.... If the water table ever drops below 10m below the topography, you are adding water to the system, and if it is ever above this point, you're substracting water.   Is this really what you want?

Usually, you'd have:

- a constant recharge acting over the entire topography (`PorousFlowSink` with negative `flux_function` - of course this could vary spatially and temporally, but constant is clearly easier to parameterise!)
- seepage - this is just your `drain` but acting over the entire topography, which will remove water from any places that it accumulates
- evapotranspiration everywhere on the topography (`PorousFlowHalfCubicSink`)

These will control the porepressure so that is almost nowhere positive, and sits close to the root depth of the evapotranspiration in many places, and can be quite negative near the tops of hills.

By the way, one reason the half-cubic or half-gaussian are advantageous numerically, is because they are smooth and differentiable, unlike the straightforward PieceiwseLinearSink.  Also, the `use_mobility` might give you better understanding of what to use for `flux_function`, and the `use_relperm` might provide quicker numerical convergence.

**Reply by WilkAndy** [reply]
By the way, feel free to ring/email, @Traiwit .   I'd prefer to sort it out on this github discussion, because then future people can learn from it, but sometimes phonecalls are much more efficient!

**Reply by Traiwit** [reply]
Thanks @WilkAndy, I will try to run more scenarios to gain more understanding of these BCs (mixing 2 or all of them), and then I will probs contact you tomorrow (Note that I'm in Sydney/AU time-zone)

Note that we are running the problem in FullySat flow model for simplicity, which might give me less realistic results.

Thank you
Traiwit

**Reply by Traiwit** [reply]
Hi @WilkAndy here is some interesting finding:

Not sure if you see above, `drain` and `topol` BCs have different `pt_vals` and `multipliers` 

```
drain
pt_vals = '0 1e9'
multipliers = '0 1e9'

topol
pt_vals = '-1e9 1e9'
multipliers = '-1e9 1e9'
```

So rain4 and rain6 are pretty much the results we want, it seems like `PorousFlowSink with flux_function = -10` doesn't do much 

Now I think I should narrow down to the controlling of the `flux_function` of `topol`, I don't have a specific value for it yet, I just know for `drain` I want the surface to be dry (flux_function as high as possible while the simulation still converges, right now 1 is good for all cases so far), However, for `topol` I've been using `flux_function = 1` for the none-cave models (working fine so far), but yeh still need to figure out for this case.

![image](https://user-images.githubusercontent.com/79562439/206085246-2f2fe647-d139-45ce-ae23-a7b70c3f25ea.png)

Regards,
Traiwit
